tests/pages/firefox/new/download.py

```diff
@@ -30,5 +31,11 @@ def download_firefox(self):
     def open_other_platforms_modal(self):
         modal = Modal(self)
         self.find_element(*self._modal_link_locator).click()
-        self.wait.until(lambda s: modal.displays(self._modal_content_locator))
+        self.wait.until(lambda s: modal.displays(self._platforms_modal_content_locator))
+        return modal
+
+    def open_firefox_account_modal(self):
+        modal = Modal(self)
+        self.download_button.click()
+        self.wait.until(lambda s: modal.displays(self._account_modal_content_locator))
```

Comment:
This test currently fails for me locally, as the modal only works if I'm not signed into Firefox with an account, and is also dependent on UITour being whitelisted. Selenium runs using it's own profile, so by default UITour is always disabled in automation for anything other than prod.

I'm not quite clear if this is expected behavior or not, as the refresh button is within that modal as well (see prior comment), but if the modal is entirely dependent on UITour, I'm not sure if it's something we can easily test in automation? (making this possible would be agreat future project).

Reply:
You should get the modal if you're in Firefox 57+ and NOT signed into an account, so it's not dependent on UITour. 

But I can't get tests to run locally (and also don't really know how to write tests) so I'm not really sure what to do here. Basically I want one test that pretends to be Firefox and clicks the button to verify that the modal opens, and another that pretends to be non-Firefox and  clicks the button to verify that it goes to /thanks.

Reply:
I've reshuffled the sequence of events as you suggested, I think this should work.